install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml

```diff
@@ -1,96 +1,86 @@
-{{- if and (.Values.agent) (not .Values.preflight.enabled) }}
+{{- if and .Values.agent (not .Values.preflight.enabled) }}
 
 {{- /*  Default values with backwards compatibility */ -}}
-{{- $defaultKeepDeprecatedProbes := "true" -}}
+{{- $defaultKeepDeprecatedProbes := true -}}
 
 {{- /* Default values when 1.8 was initially deployed */ -}}
 {{- if semverCompare ">=1.8" (default "1.8" .Values.upgradeCompatibility) -}}
-{{- $defaultKeepDeprecatedProbes = "false" -}}
+  {{- $defaultKeepDeprecatedProbes = false -}}
 {{- end -}}
-
-{{- /* Workaround so that we can set the minimal k8s version that we support */ -}}
-{{- $k8sVersion := .Capabilities.KubeVersion.Version -}}
-{{- $k8sMajor := .Capabilities.KubeVersion.Major -}}
-{{- $k8sMinor := .Capabilities.KubeVersion.Minor -}}
-
-{{- if .Values.Capabilities -}}
-{{- if .Values.Capabilities.KubeVersion -}}
-{{- if .Values.Capabilities.KubeVersion.Version -}}
-{{- $k8sVersion = .Values.Capabilities.KubeVersion.Version -}}
-{{- if .Values.Capabilities.KubeVersion.Major -}}
-{{- $k8sMajor = toString (.Values.Capabilities.KubeVersion.Major) -}}
-{{- if .Values.Capabilities.KubeVersion.Minor -}}
-{{- $k8sMinor = toString (.Values.Capabilities.KubeVersion.Minor) -}}
-{{- end -}}
-{{- end -}}
-{{- end -}}
-{{- end -}}
-{{- end -}}
```

Comment:
I haven't a real review yet, but this caught my eye - please do not remove these. They are here on purpose.

All `.Capabilities.KubeVersion` checks must be overwritable via `.Values.Capabilities.KubeVersion`, see https://github.com/cilium/cilium/pull/14778

Reply:
Hello @gandro 

`.Values.Capabilities.KubeVersion` was used to generate `quick-install.yaml`, but since cilium 1.10 `MIN_K8S_VERSION` is not using in `install/kubernetes/Makefile` anymore.
Furthermore, since 1.6, `helm template` support set Kubernetes version used for Capabilities.KubeVersion by using [`--kube-version string`](https://helm.sh/docs/helm/helm_template/). Do you think it's safe to remove this workaround?

/cc @aanm

Reply:
That's a fair point. I think we have a few users using `helm template`, so it's not just the quick-install.yaml - but I wasn't aware of the feature in Helm 3.6, that will be a good replacement going forward.

But I'll let @aanm chime in here as well. Removing Helm values is usually a breaking change, so we if we remove this, we want to communicate this clearly.

Reply:
@dungdm93 @gandro For reference, the commit that introduced this change was https://github.com/cilium/cilium/commit/4cc699dd8d99877b8456c9246ab64a741bee7030. So yeah it looks that it is fine to remove these lines as long we update the requirement of helm to >= 3.6.

Reply:
Thank @aanm, I will make this change right after #16795 is merged